src/librustdoc/clean/utils.rs

```diff
@@ -521,3 +521,19 @@ crate fn find_nearest_parent_module(tcx: TyCtxt<'_>, def_id: DefId) -> Option<De
         None
     }
 }
+
+/// Checks for the existence of `hidden` in the attribute below if `flag` is `sym::hidden`:
+///
+/// ```
+/// #[doc(hidden)]
+/// pub fn foo() {}
+/// ```
+///
+/// This function exists because it runs on `hir::Attributes` whereas the other is a
+/// `clean::Attributes` method.
+crate fn has_doc_flag(attrs: Attributes<'_>, flag: Symbol) -> bool {
```

Comment:
nit: This is ambiguous to the reader (even though the compiler knows it's from `rustc_middle`) because there's also `clean::Attributes`.
```suggestion
crate fn has_doc_flag(attrs: ty::Attributes<'_>, flag: Symbol) -> bool {
```

Reply:
Done as well.